### PR TITLE
Refactor and cleanup

### PR DESCRIPTION
**Pull request by TamimiGitHub**
A quick refactor and cleanup. 
TODO: scene tree parsing method to avoid code repetition

### CODE REVIEW

File: generator.rb [hunk 1]
```diff
@@ -46,13 +26,15 @@
         else
           puts "Unique Locator"
           #TODO: A method should be created to replace characters
-          out_file.puts("  " + node['name'].upcase.tr(" ", "_").tr("-", "_").tr(":", "") + " = { name: '" + node['name'] + "' }")
+          file.puts("  " + node['name'].upcase.tr(" ", "_").tr("-", "_").tr(":", "") + " = { name: '" + node['name'] + "' }")
```
Comment (TamimiGitHub) [inline generator.rb]: We need to do better xml parsing here

File: generator.rb [hunk 2]
```diff
@@ -2,40 +2,20 @@
...
-    out_file.puts("  # Locators")
+def get_driver
+  raise if !File.directory?('appium.txt')
+  caps = Appium.load_appium_txt file:File.join(Dir.pwd, 'appium.txt')
+  Appium::Driver.new(caps, true)
+end
+
+def add_header file, filename
+  file.puts("#!/usr/bin/env ruby")
+  file.puts("require_relative '../../../../Test/appium/common/page_helper'")
```
Comment (simongranger) [inline generator.rb:13]: The path defined here should probably be a CONST

File: generator.rb [hunk 3]
```diff
@@ -46,13 +26,15 @@
         else
           puts "Unique Locator"
           #TODO: A method should be created to replace characters
-          out_file.puts("  " + node['name'].upcase.tr(" ", "_").tr("-", "_").tr(":", "") + " = { name: '" + node['name'] + "' }")
```
Comment (simongranger) [inline generator.rb:49]: Should have a method that takes a string and returns it with all these characters replaced.

Comment (simongranger) [reply generator.rb:49]: tr(" ", "_").tr("-", "_").tr(":", "")

File: generator.rb [hunk 4]
```diff
@@ -83,17 +70,36 @@
           #@driver.find_elements(name: node['name'])
         else
           puts "Unique Locator"
-          out_file.puts("  def click_" + node['name'].tr(" ", "_").tr("-", "_").tr(":", ""))
-          out_file.puts("    click " + node['name'].upcase.tr(" ", "_").tr("-", "_").tr(":", "") + "")
-          out_file.puts("  end")
+          var_name = convert_name(node['name'])
+          file.puts("  def click_#{var_name}")
```
Comment (simongranger) [inline generator.rb:74]: var_name.downcase

File: generator.rb [hunk 5]
```diff
@@ -63,17 +50,17 @@
           #@driver.find_elements(name: node['name'])
         else
           puts "Unique Locator"
-          out_file.puts("  def " + node['name'].tr(" ", "_").tr("-", "_").tr(":", "") + "_displayed?")
-          out_file.puts("    is_displayed? " + node['name'].upcase.tr(" ", "_").tr("-", "_").tr(":", "") + "")
-          out_file.puts("  end")
+          var_name = convert_name(node['name'])
+          file.puts("  def #{var_name}_displayed?")
```
Comment (simongranger) [inline generator.rb:54]: var_name.downcase

File: generator.rb [hunk 6]
```diff
@@ -63,17 +50,17 @@
...
       end
     end
-    
-    out_file.puts("")
-    
-    # Going through all nodes that are hittable to make a 'click' method
-    out_file.puts("  # Click methods")
+end
+
+def populate_click_methods file, doc
```
Comment (simongranger) [inline generator.rb]: add driver as a parameter

File: generator.rb [hunk 7]
```diff
@@ -2,40 +2,24 @@
...
+  file.puts("#!/usr/bin/env ruby")
+  file.puts("require_relative '../../../../Test/appium/common/page_helper'")
+  file.puts("class "+  File.basename(filename,File.extname(filename)).capitalize + " < Page")
+end
+
+def convert_name node_name
+  node_name.tr(" ", "_").tr("-", "_").tr(":", "")
+end
+
+def populate_locators file, doc
```
Comment (simongranger) [inline generator.rb]: Add driver as parameter

File: generator.rb [hunk 8]
```diff
@@ -46,13 +30,16 @@
...
+          file.puts("  #{var_name.upcase} = { name: '#{node['name']}' }")
         end
       end
     end
-    out_file.puts("")
+    file.puts("")
+end
 
-    out_file.puts("  # Displayed methods")
+def populate_display_methods file, doc
```
Comment (simongranger) [inline generator.rb]: Add driver as parameter

File: generator.rb [hunk 9]
```diff
@@ -2,40 +2,24 @@
...
+
+def convert_name node_name
+  node_name.tr(" ", "_").tr("-", "_").tr(":", "")
+end
+
+def populate_locators file, doc
+  file.puts("\n\n# Locators")
     doc.xpath('//AppiumUAT//*').each do |node|
       if node['hittable'] === 'true' || node['displayed'] === 'true'
         elements = @driver.find_elements(name: node['name'])
```
Comment (simongranger) [inline generator.rb]: Replace @driver with driver everwhere it's used

File: generator.rb [hunk 10]
```diff
@@ -83,17 +70,36 @@
...
+    filename = gets.chomp.downcase
+    break if filename == "exit"
+
+    doc = Nokogiri::XML(driver.driver.page_source)
+    dirname = 'generated'
+    FileUtils.mkdir_p(dirname) if !File.directory?(dirname)
+    page_file = File.new("generated/#{filename}", "w")
+    add_header(page_file, filename)
 
+    populate_locators(page_file, doc)
```
Comment (simongranger) [inline generator.rb]: Add driver as a parameter

File: generator.rb [hunk 11]
```diff
@@ -83,17 +70,36 @@
...
+    populate_display_methods(page_file, doc)
```
Comment (simongranger) [inline generator.rb]: Add driver as a parameter

File: generator.rb [hunk 12]
```diff
@@ -83,17 +70,36 @@
...
+    populate_click_methods(page_file, doc)
```
Comment (simongranger) [inline generator.rb]: Add driver as a parameter

**Review by simongranger — CHANGES_REQUESTED** [review]

**Review by simongranger — APPROVED** [review]

**Pull request merged**